csharp/extractor/Semmle.Extraction.CSharp.Standalone/DotNet.cs

```diff
@@ -76,23 +68,33 @@ public bool AddPackage(string folder, string package)
             return RunCommand(args);
         }
 
-        public IList<string> GetListedRuntimes()
+        public IList<string> GetListedRuntimes() => GetListed("--list-runtimes", "runtime");
+
+        public IList<string> GetListedSdks() => GetListed("--list-sdks", "SDK");
+
+        private IList<string> GetListed(string args, string artifact)
         {
-            const string args = "--list-runtimes";
             progressMonitor.RunningProcess($"{dotnet} {args}");
             var pi = new ProcessStartInfo(dotnet, args)
             {
                 RedirectStandardOutput = true,
                 UseShellExecute = false
             };
-            var exitCode = pi.ReadOutput(out var runtimes);
+            var exitCode = pi.ReadOutput(out var artifacts);
             if (exitCode != 0)
             {
                 progressMonitor.CommandFailed(dotnet, args, exitCode);
                 return new List<string>();
             }
-            progressMonitor.LogInfo($"Found runtimes: {string.Join("\n", runtimes)}");
-            return runtimes;
+            progressMonitor.LogInfo($"Found {artifact}s: {string.Join("\n", artifacts)}");
+            return artifacts;
+        }
+
+        public bool Exec(string execArgs)
+        {
+            // TODO: we might need to swallow the stdout of the started process to not pollute the logs of the extraction.
+            var args = $"exec {execArgs}";
+            return RunCommand(args);
```

Comment:
This is what might be causing the failing integration test on windows.
According to the (limited information here) we probably need to avoid invoking the command as a "shell execute": https://stackoverflow.com/questions/19306194/process-launch-exception-filename-or-extension-is-too-long-can-it-be-caused-b

It would be worth trying to start the process in the same way as in the `GetListed` method (`UseShellExecute = false`) and then also fix the TODO comment and swallow the standard out by re-directing it.